docs/docs/icicle/integrations.md

```diff
@@ -0,0 +1,97 @@
+# ICICLE integrated provers
+
+ICICLE has been used by companies and projects such as [Celer Network](https://github.com/celer-network), [Consensys Gnark](https://github.com/Consensys/gnark), [EZKL](https://blog.ezkl.xyz/post/acceleration/) and others to accelerate their ZK proving pipeline.
+
+Many of these integrations have been a collaboration between Ingonyama and the integrating company. We have learned a lot about designing GPU based ZK provers.
+
+If you're interested in understanding these integrations better or learning how you can use ICICLE to accelerate your existing ZK proving pipeline this is the place for you.
+
+## A primer to building your own integrations
+
+Lets illustrate an ICICLE integration, so you can understand the core API and design overview of ICICLE.
+
+![ICICLE architecture](../../static/img/architecture-high-level.png)
+
+Engineers usually use a cryptographic library to implement their ZK protocols. These libraries implement efficient primitives which are used as building blocks for the protocol; ICICLE is such a library. The difference is that ICICLE is designed from the start to run on GPUs; the Rust and Golang APIs abstract away all low level CUDA details. Our goal was to allow developers with no GPU experience to quickly get started with ICICLE.
+
+A developer may use ICICLE with two main approaches in mind.
+
+1. Drop-in replacement approach.
+2. Full GPU replacement approach.
```

Comment:
Full GPU Replacement is confusing and sounds like we are replacing the GPU.  End-to-End Prover is cleaner.

Reply:
fix this